File: carrito_v4/src-gen/android/bd/dao/ImageProductDAO.java
//Start of user code imports
import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import java.util.ArrayList;
import java.util.List;
//End of user code
	
public class ImageProductDAO {

	private MySQLiteHelper mySQLiteHelper;

		public ImageProductDAO(MySQLiteHelper mySQLiteHelper) {
			this.mySQLiteHelper = mySQLiteHelper;
		}

		// Adding new ImageProduct
		public void addImageProduct(ImageProduct imageProduct) {
			SQLiteDatabase db = mySQLiteHelper.getWritableDatabase();

			ContentValues values = new ContentValues();
			values.put(ImageProductTable.COLUMN_IDIMAGEPRODUCT, imageProduct.getIdImageProduct());		
			values.put(ImageProductTable.COLUMN_IMAGE, imageProduct.getImage());		
			
			// Inserting Row
			db.insert(ImageProductTable.TABLE_NAME, null, values);
			db.close(); // Closing database connection
		}

		// Getting single ImageProduct    
		public ImageProduct getImageProduct(int id) {   
			SQLiteDatabase db = mySQLiteHelper.getReadableDatabase();

			ImageProduct imageProduct = new ImageProduct();

			Cursor cursor = db.query(ImageProductTable.TABLE_NAME, new String[] {
					ImageProductTable.COLUMN_IDIMAGEPRODUCT, ImageProductTable.COLUMN_IMAGE
					},		
					new String[] { String.valueOf(id) }, null, null, null, null);
			
			if (cursor != null) {
				cursor.moveToFirst();
				
				imageProduct = new ImageProduct(
						cursor.getString(0), cursor.getString(1)				
				); 

				cursor.close();
			}

			return imageProduct;
		}


			// Getting All ImageProduct
			public List<ImageProduct> getAllImageProduct() {
				SQLiteDatabase db = mySQLiteHelper.getReadableDatabase();

				List<ImageProduct> imageProductList = new ArrayList<ImageProduct>();
				// Select All Query
				String selectQuery = "SELECT  * FROM " + ImageProductTable.TABLE_NAME;

				Cursor cursor = db.rawQuery(selectQuery, null);

				// looping through all rows and adding to list
				if (cursor.moveToFirst()) {
					do {
						ImageProduct imageProduct = new ImageProduct(
						cursor.getString(1), cursor.getString(2)
						); 
						
						// Adding imageProduct to list
						 imageProductList.add( imageProduct);
					} while (cursor.moveToNext());
				}

				cursor.close();

				//return imageProduct list
				return imageProductList;
			}



			// Getting imageProduct Count
			public int getImageProductCount() {
				SQLiteDatabase db = mySQLiteHelper.getReadableDatabase();

				String countQuery = "SELECT  * FROM " + ImageProductTable.TABLE_NAME;
				Cursor cursor = db.rawQuery(countQuery, null);

				int count = cursor.getCount();

				cursor.close();

				// return count
				return count;
			}


			// Updating single imageProduct
			public int updateImageProduct(ImageProduct imageProduct) {
				SQLiteDatabase db = mySQLiteHelper.getWritableDatabase();

				ContentValues values = new ContentValues();
				values.put(ImageProductTable.COLUMN_IDIMAGEPRODUCT, imageProduct.getIdImageProduct());		
				values.put(ImageProductTable.COLUMN_IMAGE, imageProduct.getImage());		

				// updating row
				return db.update(ImageProductTable.TABLE_NAME, values,
						

			}

			// Deleting single imageProduct
			public void deleteImageProduct(ImageProduct imageProduct) {
				SQLiteDatabase db = mySQLiteHelper.getWritableDatabase();

				db.delete(ImageProductTable.TABLE_NAME,

				db.close();
			}

			// Deleting all imageProduct
			public void deleteAllImageProduct() {
				SQLiteDatabase db = mySQLiteHelper.getWritableDatabase();

				db.delete(ImageProductTable.TABLE_NAME, null, null);
				db.close();
			}

}

File: carrito_v4/src-gen/windows/ViewModel/ImageProductDAO.cs
//Start of user code imports
using Data.Common;
using Data.Model;
using SQLite;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
//End of user code

namespace Data.ViewModels
{
    class ImageProductDAO
    {
        private SQLiteConnection dbConn = new SQLiteConnection(MySQLiteHelper.DbPath);

        // Insert the new imageProduct in the ImageProduct table. 
        public void addImageProduct(ImageProduct imageProduct)
        {
            using (var dbConn = new SQLiteConnection(MySQLiteHelper.DbPath))
            {
                dbConn.RunInTransaction(() =>
                {
                    dbConn.Insert(imageProduct);
                });
            }
        }

		// Retrieve the specific ImageProduct from the database. 
        public ImageProduct getImageProduct(int id)
        {
            using (dbConn)
            {
				var queryResult = dbConn.Query<ImageProduct>("select * from ImageProduct where  =" + id).FirstOrDefault();
                return queryResult;
            }
        }

        // Retrieve the all ImageProduct List from the database. 
        public ObservableCollection<ImageProduct> getAllImageProduct()
        {
            using (dbConn)
            {
                List<ImageProduct> myCollection = dbConn.Table<ImageProduct>().ToList<ImageProduct>();
                ObservableCollection<ImageProduct> imageProductList = new ObservableCollection<ImageProduct>(myCollection);
                return imageProductList;
            }
        }

        // Retrieve the all ImageProduct List from the database and get the Total Rows. 
        public int getImageProductCount()
        {
            using (dbConn)
            {
                 List<ImageProduct> myCollection = dbConn.Table<ImageProduct>().ToList<ImageProduct>();
                ObservableCollection<ImageProduct> imageProductList = new ObservableCollection<ImageProduct>(myCollection);
                return imageProductList.Count;
            }
        }

        //Update existing ImageProduct 
        public void updateImageProduct(ImageProduct imageProduct)
        {
            using (dbConn)
            {
                var queryResult = dbConn.Query<ImageProduct>("select * from ImageProduct where ).FirstOrDefault();
                if (queryResult != null)
                {
					queryResult.idImageProduct = imageProduct.idImageProduct;		
					queryResult.image = imageProduct.image;		

                    dbConn.RunInTransaction(() =>
                    {
                        dbConn.Update(queryResult);
                    });
                }
            }
        }



        //Delete specific ImageProduct 
        public void deleteImageProduct(int id)
        {
            using (var dbConn = new SQLiteConnection(MySQLiteHelper.DbPath))
            {
                var queryResult = dbConn.Query<ImageProduct>("select * from ImageProduct where  =" + id).FirstOrDefault();

                if (queryResult != null)
                {
                    dbConn.RunInTransaction(() =>
                    {
                        dbConn.Delete(queryResult);
                    });
                }
            }
        }
        //Delete all imageProductList or delete ImageProduct table 
        public void deleteAllImageProduct()
        {
            using (var dbConn = new SQLiteConnection(MySQLiteHelper.DbPath))
            {
                //dbConn.RunInTransaction(() => 
                //   { 
                dbConn.DropTable<ImageProduct>();
                dbConn.CreateTable<ImageProduct>();
                dbConn.Dispose();
                dbConn.Close();
                //}); 
            }
        }
    }

}
